xgcm/grid_ufunc.py

```diff
@@ -335,30 +362,83 @@ def apply_as_grid_ufunc(
             for ax, width in boundary_width.items()
         }
 
-        args = grid.pad(
+        padded_args = grid.pad(
             *args,
             boundary_width=boundary_width_real_axes,
             boundary=boundary,
             fill_value=fill_value,
         )
+    else:
+        boundary_width_real_axes = {
+            real_ax: (0, 0) for real_ax in dummy_to_real_axes_mapping.values()
+        }
+        padded_args = args
+
+    if (
+        any(
+            _has_chunked_core_dims(arg, core_dims)
+            for arg, core_dims in zip(args, in_core_dims)
+        )
+        and map_overlap
+    ):
+        # map operation over dask chunks along core dimensions
+        from dask.array import map_overlap
+
+        # merge any lonely chunks from padding
+        rechunked_padded_args = _rechunk_to_merge_in_boundary_chunks(
+            padded_args,
+            args,
+            boundary_width_real_axes,
+            grid,
+        )
+
+        # TODO should this be adjusted?
```

Comment:
Adjusted in the sense of "do we need to adjust the chunksize when dealing with the case of the length of an axis changing during a grid ufunc operation?". Now unnecessary because we chose not to support that case.